voice-search/voice-search-impl/src/main/java/com/duckduckgo/voice/impl/VoiceSearchPermissionCheck.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.voice.impl
+
+import android.Manifest
+import android.content.Context
+import android.content.pm.PackageManager
+import androidx.core.content.ContextCompat
+import com.duckduckgo.di.scopes.ActivityScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface VoiceSearchPermissionCheck {
+    fun hasRequiredPermissionsGranted(): Boolean
+}
+
+@ContributesBinding(ActivityScope::class)
+class RealVoiceVoiceSearchPermissionCheck @Inject constructor(
+    private val context: Context
+) : VoiceSearchPermissionCheck {
+    override fun hasRequiredPermissionsGranted(): Boolean = ContextCompat.checkSelfPermission(
+        context, Manifest.permission.RECORD_AUDIO
+    ) == PackageManager.PERMISSION_GRANTED
```

Comment:
What happens if the user selects "Ask every time" in app's permission settings?

Reply:
I followed the pattern that IG / FB uses that you still need to go through settings to actually allow the feature. However, I just noticed now that when you close the feature that requests the camera / microphone (example: stories) and open it again, the permission is re-requested again.

Reply:
What I mean is, If I deny the permission and then click the microphone icon, it tells me to go to settings and allow the feature. If I choose "Ask every time" in settings and return to the app, it still tells me to go to settings to allow the permission. Is this expected behavior?

Reply:
We discussed this on zoom and decided that we will not fix this since this issue is caused by the Android permissions API and also happens on other apps.